how-to/customize-workspace/client/src/framework/shapes/integrations-shapes.ts

```diff
@@ -35,25 +36,50 @@ export interface IntegrationHelpers extends ModuleHelpers {
 	 * @param bounds The optional bounds for the page.
 	 * @returns The window created.
 	 */
-	launchPage?(page: Page, bounds?: OpenFin.Bounds): Promise<BrowserWindowModule>;
+	launchPage(page: Page, bounds?: OpenFin.Bounds): Promise<BrowserWindowModule>;
+
+	/**
+	 * Get the bounds for the page.
+	 * @param pageId The page to get the bounds for.
+	 * @param fromStorage Get the bounds from storage, default to false.
+	 */
+	getPageBounds(pageId: string, fromStorage?: boolean): Promise<OpenFin.Bounds | null>;
 
 	/**
 	 * Launch a snapshot.
 	 * @param snapshotUrl The snapshot url
 	 */
-	launchSnapshot?(snapshotUrl: string): Promise<OpenFin.Identity[]>;
+	launchSnapshot(snapshotUrl: string): Promise<OpenFin.Identity[]>;
 
 	/**
 	 * Open a url with the browser.
 	 * @param url The url to open.
 	 */
-	openUrl?(url: string): Promise<void>;
+	openUrl(url: string): Promise<void>;
 
 	/**
 	 * Set the home search query.
 	 * @param query The query to set.
 	 */
-	setSearchQuery?(query: string): Promise<void>;
+	setSearchQuery(query: string): Promise<void>;
+
+	/**
+	 * Get the value of a condition.
+	 * @param conditionId The id of the condition.
+	 * @returns True if the condition is set.
+	 */
+	condition(conditionId: string): Promise<boolean>;
+
+	/**
+	 * Share data.
+	 * @param options The sharing options.
+	 */
+	share(options?: IShareCustomData): Promise<void>;
```

Comment:
I worry that we might be adding a few too many helpers to this. Share I guess might be useful later for other sharing of integration data (but then it feels like this could be an intent especially as it is a void?).

With random uuid I know it is convenient but because it is such a small thing maybe it should just import the ts file by reference rather than extending the api?

Reply:
for randomUUID you cant just import the ts as you are referencing something that might be outside the scope of the module, same for any other methods in the framework